app/controllers/memberships_controller.rb

```diff
@@ -39,6 +39,14 @@ def accept
     Membership.find(params[:membership_id]).accept!
   end
 
+  def reject
+    membership = Membership.find(params[:membership_id])
+    Membership::WithdrawService.call(membership)
+
+  rescue Membership::WithdrawService::MembershipMustHaveDefaultPost
+    forbidden_page
+  end
+
```

Comment:
We thought about that when we implemented this feature, because these two actions shape are realy similar.
The two actions are authenticated differently and the give different responses on success.(render reject.js.erb or redirect)
I think it would be weird to figure out who sent the request (group leader or regular member) in the controller and generate the response accordingly, but tomorrow I will try it.